### DISCUSSION TITLE
I can't understand why the /plaintext test is almost 5 times faster than the /json test.

### DISCUSSION THREAD

**Discussion by remittor** [TechEmpower/FrameworkBenchmarks#8047] [Q&A] [answered]
Here are the results of the tests on my stand:

### `$ ./libreactor-server`
```
$ wrk -t1 -c100 -d15 -H 'Connection: keep-alive' http://127.0.0.1:8080/plaintext
Running 15s test @ http://127.0.0.1:8080/plaintext
  1 threads and 100 connections
  Thread Stats   Avg      Stdev     Max   +/- Stdev
    Latency   464.05us   64.37us   3.03ms   83.36%
    Req/Sec   109.56k     3.04k  116.05k    72.67%
  1634595 requests in 15.00s, 210.45MB read
Requests/sec: 108952.50
Transfer/sec:     14.03MB

$ wrk -t1 -c100 -d15 -H 'Connection: keep-alive' http://127.0.0.1:8080/json
Running 15s test @ http://127.0.0.1:8080/json
  1 threads and 100 connections
  Thread Stats   Avg      Stdev     Max   +/- Stdev
    Latency   488.96us   97.31us   1.84ms   93.74%
    Req/Sec   105.41k     4.17k  111.84k    74.67%
  1573334 requests in 15.00s, 232.57MB read
Requests/sec: 104875.40
Transfer/sec:     15.50MB
```

The executable file `libreactor-server` was obtained by me clearly following this instruction:  https://github.com/TechEmpower/FrameworkBenchmarks/blob/master/frameworks/C/libreactor/libreactor-server.dockerfile

But I had to fix 1 line in the code to force the `libreactor-server` to use only 1 thread.

------
And here are the results for `libreactor-server` contained in database **TFB**:
plaintext: **7024777** req/sec ([proof-link](https://www.techempower.com/benchmarks/#section=test&runid=ef7991e0-35e3-4de1-94b6-1ac13e29b9c6&test=plaintext&l=zik0zi-35r))
json: **1306654** req/sec ([proof-link](https://www.techempower.com/benchmarks/#section=test&runid=ef7991e0-35e3-4de1-94b6-1ac13e29b9c6&test=json&l=zik0zi-35r))

------
```
for 56 threads: 7024777 / 1306654 = 5.37
for  1 thread :  108952 /  104875 = 1.03
```
Explain to me where such a difference in the ratio of speeds comes from.

**Comment by fakeshadow** [comment, marked as answer]
plaintext bench uses http/1.1 pipeline feature. You can checkout the script tfb uses [here](https://github.com/TechEmpower/FrameworkBenchmarks/tree/master/toolset/wrk)

**Comment by remittor** [comment]
@fakeshadow , thanks
Now it's clearer:
```
$ wrk -t1 -c100 -d15 -H 'Connection: keep-alive' http://127.0.0.1:8080/plaintext -s pipeline.lua -- 16
Running 15s test @ http://127.0.0.1:8080/plaintext
  1 threads and 100 connections
  Thread Stats   Avg      Stdev     Max   +/- Stdev
    Latency   379.06us  269.40us   3.01ms   18.70%
    Req/Sec     0.93M    36.40k    0.99M    84.00%
  13863936 requests in 15.00s, 1.74GB read
Requests/sec: 924084.24
Transfer/sec:    118.97MB
```
924084 / 104875 = 8.81

**Comment by remittor** [comment]
Now I wanted to check the correct implementation of the pipeline on this server.
To do this, I wrote a python script that sends inseparably **10M** identical requests `/plaintext`.
The result is deplorable. The `libreactor-server` simply loaded everything into RAM:
![image](https://user-images.githubusercontent.com/9350115/226026011-34159af5-9379-4320-b739-d320f88b6422.png)

The script has not been working for 10 minutes, and the `libreactor-server` does not release memory.

It seems to me that this is a common situation. And it looks like you need to create a separate topic about this.
